tracer/src/Datadog.Trace/Iast/Aspects/System.Text/StringBuilderAspects.cs

```diff
@@ -183,24 +270,34 @@ public static StringBuilder Append(StringBuilder? target, char[]? value, int sta
     [AspectMethodReplace("System.Text.StringBuilder::Append(System.Object)")]
     public static StringBuilder Append(StringBuilder? target, object? value)
     {
-        var initialLength = target?.Length ?? 0;
-
-        object? valueObject;
-        int length;
-        if (value is StringBuilder valueStringBuilder)
+        var result = target!.Append(value);
+        try
         {
-            valueObject = valueStringBuilder;
-            length = valueStringBuilder!.Length;
+            if (target is not null && value is not null)
```

Comment:
`If value is null, no changes are made.` from the [doc](https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder.append?view=net-8.0#system-text-stringbuilder-append(system-string))
Also, if it throws, it will throw in the original call.